pkg/user/password_reset_test.go

```diff
@@ -0,0 +1,100 @@
+package user
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewPasswordReset(t *testing.T) {
+	pr := NewPasswordReset()
+	assert.NotNil(t, pr)
+	assert.NotEmpty(t, pr.Token)
+	time.Sleep(1 * time.Second)
+	assert.Less(t, pr.CreatedAt.Unix(), time.Now().Unix())
```

Comment:
Don't use sleep in the test. It make unit tests slow. Consider mocking `time.Now`.

```go
var timeNow = time.Now

func NewPasswordReset() *PasswordReset {
	return &PasswordReset{
		// ...
		CreatedAt: timeNow(),
	}
}
```

```suggestion
	mockTime := time.Now()
	timeNow = func() time.Time { return mockTime }
	pr := NewPasswordReset()
	assert.NotNil(t, pr)
	assert.NotEmpty(t, pr.Token)
	assert.Equal(t, mockTime, pr.CreatedAt)
```

Reply:
@rot1024 In this particular case I could solve it without sleep nor mock